packages/libs/error-stack/src/report.rs

```diff
@@ -211,13 +259,115 @@ impl<C> Report<C> {
+    #[deprecated = "a report might contain multiple backtraces, use `request_ref::<Backtrace>()` \
+                    instead"]
+    pub fn backtrace(&self) -> Option<&Backtrace> {
+        self.request_ref::<Backtrace>().next()
+    }
+
+    #[allow(missing_docs)]
+    #[must_use]
+    #[cfg(feature = "spantrace")]
+    #[cfg_attr(
+        nightly,
+        deprecated = "a report might contain multiple spantraces, use \
+                      `request_ref::<SpanTrace>()` instead"
+    )]
+    #[cfg_attr(
+        not(nightly),
+        deprecated = "a report might contain multiple spantraces, use \
+                      `frames().filter(Frame::downcast_ref::<SpanTrace>)` instead"
+    )]
+    pub fn span_trace(&self) -> Option<&SpanTrace> {
+        #[cfg(nightly)]
+        return self.request_ref::<SpanTrace>().next();
+
+        #[cfg(not(nightly))]
+        return self.downcast_ref::<SpanTrace>();
+    }
+
+    /// Merge two [`Report`]s together
+    ///
+    /// This function appends the [`current_frames()`] of the other [`Report`] to the
+    /// [`current_frames()`] of this report.
+    /// Meaning `A.extend_one(B) -> A.current_frames() = A.current_frames() + B.current_frames()`
+    ///
+    /// [`current_frames()`]: Self::current_frames
+    ///
+    /// ```rust
+    /// use std::{
+    ///     fmt::{Display, Formatter},
+    ///     path::Path,
+    /// };
+    ///
+    /// use error_stack::{Context, Report, IntoReport, ResultExt};
+    ///
+    /// #[derive(Debug)]
+    /// struct IoError;
+    ///
+    /// impl Display for IoError {
+    ///     # fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+    ///     #     const _: &str = stringify!(
+    ///             ...
+    ///     #     );
+    ///     #     f.write_str("Io Error")
+    ///     # }
+    /// }
+    ///
+    /// # impl Context for IoError {}
+    ///
+    /// # #[allow(unused_variables)]
+    /// fn read_config(path: impl AsRef<Path>) -> Result<String, Report<IoError>> {
+    ///     # #[cfg(any(miri, not(feature = "std")))]
+    ///     # return Err(error_stack::report!(IoError).attach_printable("Not supported"));
+    ///     # #[cfg(all(not(miri), feature = "std"))]
+    ///     std::fs::read_to_string(path.as_ref())
+    ///         .into_report()
+    ///         .change_context(IoError)
+    /// }
+    ///
+    /// let mut error1 = read_config("config.txt").unwrap_err();
+    /// let error2 = read_config("config2.txt").unwrap_err();
+    /// let mut error3 = read_config("config3.txt").unwrap_err();
+    ///
+    /// error1.extend_one(error2);
+    /// error3.extend_one(error1);
+    ///
+    /// // ^ This is equivalent to:
+    /// // error3.extend_one(error1);
+    /// // error3.extend_one(error2);
+    /// ```
+    ///
+    /// This function implements the same functionality as
+    /// [`Extend::extend_one` (#7261)](https://github.com/rust-lang/rust/issues/72631).
+    /// Once stabilised this function will be deprecated in favor of [`Extend`].
```

Comment:
I guess it will be removed?
```suggestion
    /// Once stabilised this function will be removed in favor of [`Extend`].
```